alchemy/src/esbuild/bundle.ts

```diff
@@ -119,41 +119,53 @@ export const Bundle = Resource(
     props: BundleProps
   ): Promise<Bundle> {
     // Determine output path
-    const outputPath = getOutputPath(props);
+    const outDirPath = getOutDirPath(props);
 
     // Ensure output directory exists
-    await fs.promises.mkdir(path.dirname(outputPath), { recursive: true });
-
+    await fs.promises.mkdir(path.dirname(outDirPath), { recursive: true });
     if (this.phase === "delete") {
-      await fs.promises.unlink(outputPath).catch(() => {});
-      // Also clean up sourcemap if it exists
-      await fs.promises.unlink(outputPath + ".map").catch(() => {});
+      await cleanDirectory(outDirPath);
       return this.destroy();
     }
 
     const result = await bundle(props);
+    // Check that bundle created output an file for the given entrypoint
+    // Use bundle metada data to retrieve its content
+    let bundleOutputPath;
+    for (const file in result.metafile.outputs) {
+      if (result.metafile.outputs[file].entryPoint === props.entryPoint) {
+        bundleOutputPath = file;
+      }
+    }
+    if (!bundleOutputPath) {
+      throw new Error(`Unable to find a compiled file`);
+    }
```

Comment:
I don't understand why this is necessary? Can't we deterministically know where it will place the bundle?

How sure are we that `result.metafile.outputs[file].entryPoint === props.entryPoint` is reliable and won't be messed up by esbuild adding a slash or something like that?

Perhaps it can be simplified to?
```ts
const bundleOutputPath = Object.entries(result.metafile.outputs)
  .find(([file, output]) => output.entryPoint === props.entryPoint)
  ?.[0];
```

Reply:
the issue is that if you were to rely on props as before, but let say esbuild change it to .mjs, then you might not able to find the file, I would try to use bundle output information as much as possible vs rely on inputs and go chase it. it should be more robust like this, but yeah can totally simplify.

Reply:
Let me also test for if this works well with source maps (output files >1 )

Reply:
Might be worth adding a test with that too in `function.test.ts`